libvast/src/evaluate.cpp

```diff
@@ -27,6 +27,12 @@ namespace vast {
 
 namespace {
 
+template <class LhsView, class RhsView>
+inline constexpr auto requires_stdcmp
+  = std::is_integral_v<LhsView> && std::is_integral_v<RhsView>
+    && !std::is_same_v<LhsView, RhsView>
+    && !detail::is_any_v<bool, LhsView, RhsView>;
+
```

Comment:
Any reason not to use `std::cmp_*` unconditionally?

Reply:
It only works for integral but non-bool types. With the removal of `vast.:integer` I got `-Wsign-compare` warnings from clang here, so I had to fix it given that our CI is using `-Werror`.